packages/backend-function/src/factory.ts

```diff
@@ -209,14 +222,43 @@ class AmplifyFunction
   readonly resources: FunctionResources;
   constructor(scope: Construct, id: string, props: HydratedFunctionProps) {
     super(scope, id);
+    const functionLambda = new NodejsFunction(scope, `${id}-lambda`, {
+      entry: props.entry,
+      environment: props.environment as { [key: string]: string }, // for some reason TS can't figure out that this is the same as Record<string, string>
+      timeout: Duration.seconds(props.timeoutSeconds),
+      memorySize: props.memoryMB,
+      runtime: nodeVersionMap[props.runtime],
+      bundling: {
+        // Added '\' to the end of each line for readability here
+        // eslint-disable-next-line spellcheck/spell-checker
+        banner: `import { SSM } from '@aws-sdk/client-ssm'; const client = new SSM();\
+        const envArray = ${JSON.stringify(Object.keys(props.environment))};\
```

Comment:
Once this snippet is moved to it's own file, we may need to store this list as a separate env var like `SECRET_NAME_LIST=["VAR_1", "VAR_2", 'etc']` that we can read to determine which secrets to look up.

Also, not every environment key is a secret.

Reply:
Gotcha. And yeah not every environment key is a secret, `getParameters` will list them as `InvalidParameters` in the response while the env vars that are secret paths will have their results in `Parameters`.

Reply:
We shouldn't rely on that as an implicit signal. While unlikely, someone could have some other parameter path stored in an env var that shouldn't be included here. We should only call SSM for the secret paths that we care about.